packages/@react-aria/dnd/src/DragManager.ts

```diff
@@ -463,11 +463,9 @@ class DragSession {
       });
     }
 
-    // Blur and re-focus the drop target so that the focus ring appears.
-    if (this.currentDropTarget) {
-      this.currentDropTarget.element.blur();
-      this.currentDropTarget.element.focus();
-    }
+    // Blur and re-focus the drag target so that the focus ring appears.
+    this.dragTarget.element.blur();
+    this.dragTarget.element.focus();
```

Comment:
Did some poking around and noticed that this change breaks old functionality, namely keyboard drag and drop from a list to a drop zone or dropping between two lists in the dnd stories. The old behavior shifted focus to the dropped item but this change seems to make it so the previously dragged item keeps focus.

older storybooks:
https://reactspectrum.blob.core.windows.net/reactspectrum/b4ea4c93db98dfea5e2caac389dee12c486251fc/storybook/index.html?path=/story/listview-drag-and-drop--drag-out-of-list&providerSwitcher-toastPosition=bottom
https://reactspectrum.blob.core.windows.net/reactspectrum/b4ea4c93db98dfea5e2caac389dee12c486251fc/storybook/index.html?path=/story/drag-and-drop--droppable-grid-with-many-items&providerSwitcher-toastPosition=bottom

new experience:
https://reactspectrum.blob.core.windows.net/reactspectrum/b194c264b06bbec770cf44d6ffd99ae55b62f658/storybook/index.html?path=/story/listview-drag-and-drop--drag-out-of-list&providerSwitcher-toastPosition=bottom
https://reactspectrum.blob.core.windows.net/reactspectrum/b194c264b06bbec770cf44d6ffd99ae55b62f658/storybook/index.html?path=/story/drag-and-drop--droppable-grid-with-many-items&providerSwitcher-toastPosition=bottom

I feel like this isn't the right place to make this change anyways, maybe somewhere in `useDroppableCollection` since it seems to handle cases similar to this? [This area of code](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/dnd/src/useDroppableCollection.ts#L186-L197) doesn't seem to get hit at all in the reorderable example, probably because it doesn't actually increase the collection size since it is a reorder example... It also doesn't get hit when keyboard dragging and dropping items between two ListViews because the ListView that receives the drop doesn't receive focus... That block of code does get hit when performing the drag and drop operation via mouse so it would be good to figure why the behaviors are different there. Similar observation for keyboard vs mouse reordering, focus is properly set when reordering via the mouse

https://github.com/adobe/react-spectrum/blob/a0a88da217db1cff67be1bdc94490b3a7002b6f3/packages/%40react-aria/dnd/src/useDroppableCollection.ts#L136-L153 would've technically handled the inter-ListView drag and drop behavior if we strip out the `target.dropPosition === 'on'` but I'm wary of changing that behavior there

Reply:
Happy to take a look at this with you together as well

Reply:
Great find, I'm taking a look at other options. I'll probably close this out and make a new PR if the solution ends up being largely different